src/v0/destinations/factorsai/transform.js

```diff
@@ -13,6 +13,15 @@ const { JSON_MIME_TYPE } = require('../../util/constant');
 
 const { ConfigCategories, mappingConfig, BASE_URL } = require('./config');
 
+function populateIpDetails(requestJson, message) {
+  const payload = requestJson;
+  if (message.context && requestJson.context) {
```

Comment:
I don't feel this condition is right. May be this `payload.context && (message.context || message.request_ip)` will work better for us

But even below way of rewriting the logic might be good for us
```javascript
if (message.context || message.request_ip) {
  // assuming `payload.context` is an object
  payload.context = { ...(payload.context || {}), ip: message.context?.ip || message.request_ip }
}
```
Please do evaluate this before making changes

Reply:
updated, added more test case

Reply:
I think we should consider the condition where `payload.context` can be anything other than object. Apart from that I'm good with the changes

Reply:
context according to rudderstack spec is an object. Hence closing this comment

Reply:
it must be an object as per rudder contract